src/MongoDB.Driver.Core/Core/Connections/BinaryConnection.cs

```diff
@@ -738,6 +743,23 @@ private Exception WrapException(Exception ex, string action)
             }
         }
 
+        private Exception WrapInOperationCanceledExceptionIfRequired(Exception exception, CancellationToken cancellationToken)
+        {
+            if (exception is ObjectDisposedException objectDisposedException
+                &&
+                (
+                    objectDisposedException.ObjectName == GetType().Name ||
+                    objectDisposedException.Message == "The semaphore has been disposed."
```

Comment:
These two lines (specially the second one) **might** be platform dependent. It **might** also be language dependent (not sure if Exception error messages are always in English).

Just something to keep in mind if we get reports of any issues.

Reply:
Discussed offline. It's true that the message is language-specific. So additional conditions for objectDisposedException are not good enough here. Removed.